15/slim/debian/Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM debian:buster-slim
+
+ARG version=15.0.0.36-1
+# In addition to installing the Amazon corretto, we also install
+# fontconfig. The folks who manage the docker hub's
+# official image library have found that font management
+# is a common usecase, and painpoint, and have
+# recommended that Java images include font support.
+#
+# See:
+#  https://github.com/docker-library/official-images/blob/master/test/tests/java-uimanager-font/container.java
+#
+# Slim:
+#   JLink is used (retaining all modules) to create a slimmer version of the JDK excluding man-pages, header files and debugging symbols - saving ~113MB.
+RUN set -eux \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        curl ca-certificates gnupg software-properties-common fontconfig java-common \
+    && curl -fL https://apt.corretto.aws/corretto.key | apt-key add - \
```

Comment:
I am aware that this is the official method recommended by Amazon however, this adds the key in manner which grants it validity over any software repository, including those not distributed by Amazon. I understand this is quite pedantic but for illustrative purposes, the most correct and idiomatic way to add repositories on Debian is to add the following two apt config files:

`/etc/apt/sources.list.d/apt.coretto.aws.sources`

```
Types: deb
URIs: https://apt.corretto.aws
Suites: stable
Components: main
Signed-By: /usr/local/share/keyrings/apt.corretto.aws.gpg
```

And something like:

`/etc/apt/preferences.d/apt.coretto.aws.pref`
```
Package: *
Pin: origin apt.corretto.aws
Pin-Priority: 1

Package: java-1.8.0-amazon-corretto-jdk
Pin: origin apt.corretto.aws
Pin-Priority: 500

Package: java-11-amazon-corretto-jdk
Pin: origin apt.corretto.aws
Pin-Priority: 500

Package: java-15-amazon-corretto-jdk
Pin: origin apt.corretto.aws
Pin-Priority: 500
```

And a gpg de-armored key here `/usr/local/share/keyrings/apt.corretto.aws.gpg`. The parent directory for this file will need to be created but it is a more FHS-correct than `/usr/share/keyrings` which you may be tempted to use.

See: https://wiki.debian.org/DebianRepository/UseThirdParty